src/main/java/com/commercetools/sync/commons/utils/QueryAll.java

```diff
@@ -0,0 +1,106 @@
+package com.commercetools.sync.commons.utils;
+
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.queries.PagedQueryResult;
+import io.sphere.sdk.queries.QueryDsl;
+import io.sphere.sdk.queries.QuerySort;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static java.util.stream.Collectors.toList;
+
+final class QueryAll<T, C extends QueryDsl<T, C>> {
```

Comment:
Since it was not meant except  for internal use of this library, no javadocs were added. But now I added anyways to make it a little bit clearer: 106f41f